### ISSUE TITLE
Cannot create keybindings for multiple-cursors

### ISSUE THREAD

**Issue by redpanda1234** [raxod502/radian#345]
Saw that in radian-formatting.el had a use-package multiple-cursors delayed install line.  Can't figure out how to get it configured?  I added keyboard bindings to my init file, but upon calling mc/edit-lines on a highlighted region, I'm getting a "Wrong type argument: commandp" error?

**Comment by redpanda1234** [comment]
# ;_;

**Comment by raxod502** [comment]
This will be because multiple-cursors uses deferred installation, and creating keybindings for lazy-installed packages requires more care than usual. Specifically, once a package is installed, any of its autoloaded functions (usually packages will autoload all of their public functions and interactive commands) will be available, and calling any of them will load the package. However, this is only possible if the package is already installed, since otherwise there is no way for Emacs to know what autoloads are provided by the package. As a result, for calling a command to trigger deferred installation, that command needs to be manually specified. Referencing it from the `use-package` form will accomplish this, so you may want to consider establishing your keybindings via

```
(use-package multiple-cursors
  :defer-install t
  :bind ((mc/edit-lines . "something")
          ;; more keybindings
         ))
```

Alternatively, you can skip the whole deferred installation issue by simply installing the package eagerly, i.e.

```
(use-package multiple-cursors)
```

AFAIK, `use-package` forms can be safely repeated, so you can just put this stuff in `init.local.el`.

The only reason I put `multiple-cursors` there was because it was already loaded as a dependency, and I needed to update the recipe (although this can now be done via `straight-register-package`). I didn't actually configure it properly for deferred installation.

I'll replace the `use-package` declaration with a `straight-register-package`, but in any case let me know if the above solutions work.

**Comment by raxod502** [comment]
If you give the specific code you're using to configure multiple-cursors I can translate that into `use-package` format.

**Comment by raxod502** [comment]
This issue is being closed automatically since it has a "waiting on response" label but hasn't had any activity in 90 days. But don't worry: if you have some information to help move the discussion along, just leave a comment and I'll re-open the issue.

*(Comment generated automatically by [NO CARRIER](https://github.com/twbs/no-carrier) via [tidier](https://github.com/raxod502/tidier).)*
